clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1171,4 +1253,43 @@ private void release() {
         if (refcount.decrementAndGet() == 0)
             currentThread.set(NO_CURRENT_THREAD);
     }
+
+    private class PatternSubscriptionTask implements DelayedTask {
+        private final long interval;
+
+        public PatternSubscriptionTask(long interval) {
+            this.interval = interval;
+        }
+
+        public void run(final long now) {
+            final List<String> topicsToSubscribe = new ArrayList<>();
+
+            final RequestFuture<ClientResponse> requestFuture = fetcher.sendMetadataRequest();
+            if (requestFuture != null) {
+                requestFuture.addListener(new RequestFutureListener<ClientResponse>() {
+                    @Override
+                    public void onSuccess(ClientResponse value) {
+                        MetadataResponse response =
```

Comment:
Do you mean in case of errors, I should log it and schedule the task as in case of onFailure?